src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ExternalContracts.cs

```diff
@@ -9,6 +9,8 @@
 using NuGet.SolutionRestoreManager;
 using NuGet.VisualStudio;
 
+#pragma warning disable RS0030 // Do not used banned APIs
```

Comment:
Should there be a `#pragma warning restore RS0030` somewhere? I assume it doesn't really matter in this file.

Reply:
I don't think it's necessary in this file. So long as we only ever model external contracts here.

This file's just adding metadata to external contracts, which is still useful, even if we have banned their use.

Most of these banned contracts are actually used indirectly, where we wrap them in an unbanned API that gives some guarantee we want to have, and suppress RS0030 on that usage.